test/e2e_test.go

```diff
@@ -339,117 +391,134 @@ func TestOCIStorage(t *testing.T) {
+				"artifacts.pipelinerun.signer":    "x509",
+				"artifacts.pipelinerun.storage":   "tekton,oci",
+				"storage.oci.repository.insecure": "true",
+			},
+			getObject: kanikoPipelineRun,
+		},
 	}
 
-	taskRun := kanikoTaskRun(ns)
-	tr, err := c.PipelineClient.TektonV1beta1().TaskRuns(ns).Create(ctx, taskRun, metav1.CreateOptions{})
-	if err != nil {
-		t.Errorf("error creating taskrun: %s", err)
-	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			image := "chains-test-multibackendstorage"
+			ctx := logtesting.TestContextWithLogger(t)
+			c, ns, cleanup := setup(ctx, t, setupOpts{
+				registry:        true,
+				kanikoTaskImage: image,
+			})
+			defer cleanup()
 
-	publicKey, err := cryptoutils.MarshalPublicKeyToPEM(c.secret.x509priv.Public())
-	if err != nil {
-		t.Error(err)
-	}
-	verifyTaskRun := verifyKanikoTaskRun(ns, image, string(publicKey))
-	tr, err = c.PipelineClient.TektonV1beta1().TaskRuns(ns).Create(ctx, verifyTaskRun, metav1.CreateOptions{})
-	if err != nil {
-		t.Fatalf("error creating task: %s", err)
-	}
+			resetConfig := setConfigMap(ctx, t, c, test.cm)
+			defer resetConfig()
+			time.Sleep(3 * time.Second)
 
-	// Give it a minute to complete.
-	waitForCondition(ctx, t, c.PipelineClient, tr.Name, ns, signed, 60*time.Second)
+			createdObj := tekton.CreateObject(t, ctx, c.PipelineClient, test.getObject(ns))
 
-	// It can take up to a minute for the secret data to be updated!
-	tr = waitForCondition(ctx, t, c.PipelineClient, tr.Name, ns, signed, 120*time.Second)
+			publicKey, err := cryptoutils.MarshalPublicKeyToPEM(c.secret.x509priv.Public())
+			if err != nil {
+				t.Error(err)
+			}
 
-	// Verify the payload signature.
-	verifySignature(ctx, t, c, tr)
+			// Verify object has been signed
+			obj := waitForCondition(ctx, t, c.PipelineClient, createdObj, signed, 120*time.Second)
 
+			// Verify the payload signature.
+			verifySignature(ctx, t, c, obj)
+
+			verifyTro := verifyKanikoTaskRun(ns, fmt.Sprintf("%s/%s", c.internalRegistry, image), string(publicKey))
+			createdVerifyTro := tekton.CreateObject(t, ctx, c.PipelineClient, verifyTro)
+			waitForCondition(ctx, t, c.PipelineClient, createdVerifyTro, successful, 60*time.Second)
+		})
+	}
 }
 
 func TestRetryFailed(t *testing.T) {
-	ctx := logtesting.TestContextWithLogger(t)
-	c, ns, cleanup := setup(ctx, t, setupOpts{registry: true})
-	defer cleanup()
+	tests := []struct {
+		name      string
+		cm        map[string]string
+		opts      setupOpts
+		getObject func(ns string) objects.TektonObject
+	}{
+		{
+			name: "taskrun",
+			cm: map[string]string{
+				// don't set insecure repository, forcing signature upload to fail
+				"artifacts.oci.storage":     "oci",
+				"artifacts.taskrun.storage": "tekton",
+				"storage.oci.repository":    "gcr.io/not-real",
+			},
+			opts: setupOpts{
+				registry:        true,
+				kanikoTaskImage: "chains-test-tr-retryfailed",
+			},
+			getObject: getTaskRunObject,
+		},
+	}
```

Comment:
Should there be a pipelinerun test here as well?

Reply:
Yes, something i was going to come back to but slipped by. I added the test.